src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs

```diff
@@ -81,20 +81,58 @@ public string GetDumpFile()
         }
 
         /// <inheritdoc/>
-        public void StartProcessDump(int processId, string dumpFileGuid, string testResultsDirectory, bool isFullDump = false)
+        public void StartTriggerBasedProcessDump(int processId, string dumpFileGuid, string testResultsDirectory, bool isFullDump = false)
         {
             this.dumpFileName = $"{this.processHelper.GetProcessName(processId)}_{processId}_{dumpFileGuid}";
             this.testResultsDirectory = testResultsDirectory;
 
+            string procDumpArgs = new ProcDumpArgsBuilder().BuildTriggerBasedProcDumpArgs(
+                processId,
+                this.dumpFileName,
+                ProcDumpExceptionsList,
+                isFullDump);
+
+            if (EqtTrace.IsInfoEnabled)
+            {
+                EqtTrace.Info($"ProcessDumpUtility : The procdump argument is {procDumpArgs}");
+            }
+
             this.procDumpProcess = this.processHelper.LaunchProcess(
                                             this.GetProcDumpExecutable(processId),
-                                            ProcessDumpUtility.BuildProcDumpArgs(processId, this.dumpFileName, isFullDump),
+                                            procDumpArgs,
                                             testResultsDirectory,
                                             null,
                                             null,
                                             null) as Process;
         }
 
+        /// <inheritdoc/>
+        public void StartHangBasedProcessDump(int processId, string dumpFileGuid, string testResultsDirectory, bool isFullDump = false)
+        {
+            this.dumpFileName = $"{this.processHelper.GetProcessName(processId)}_{processId}_{dumpFileGuid}_hangdump";
+            this.testResultsDirectory = testResultsDirectory;
+
+            string procDumpArgs = new ProcDumpArgsBuilder().BuildHangBasedProcDumpArgs(
+                processId,
+                this.dumpFileName,
+                isFullDump);
+
+            if (EqtTrace.IsInfoEnabled)
+            {
+                EqtTrace.Info($"ProcessDumpUtility : The hang based procdump invocation argument is {procDumpArgs}");
+            }
+
+            this.procDumpProcess = this.processHelper.LaunchProcess(
+                                            this.GetProcDumpExecutable(processId),
+                                            procDumpArgs,
+                                            testResultsDirectory,
+                                            null,
+                                            null,
+                                            null) as Process;
+
+            this.procDumpProcess.WaitForExit();
```

Comment:
>this.procDumpProcess.WaitForExit(); [](start = 12, length = 35)

this is not safe.. better to be event based or periodic checks

Reply:
okay, will change